recipes/_load_databag_config.rb

```diff
@@ -143,7 +143,13 @@
 
 unmanaged_hosts = nagios_bags.get(node['nagios']['unmanagedhosts_databag'])
 unmanaged_hosts.each do |item|
-  next unless item['environment'].nil? || item['environment'] == node.chef_environment
+  if node['nagios']['multi_environment_monitoring']
+    envs = node['nagios']['monitored_environments']
+    next if item['environment'].nil? || !envs.include?(item['environment'])
+  else
+    next unless item['environment'].nil? || item['environment'] == node.chef_environment
+  end
```

Comment:
It's about making the cookbook simple-stupid or more like you-know-what-you-are-doing. Your option indeed makes more sense for the overall, so I will update the PR.